lib/cpu-profile-to-tree.js

```diff
@@ -21,8 +21,44 @@ function convert (profile) {
   return converted
 }
 
+function id2IndexMap (nodes) {
+  let map = new Array(nodes.length + 1)
+  for (let i = 0; i < nodes.length; i++) {
+    const node = nodes[i]
+    map[node.id] = i
+  }
+  return map
+}
+
+function convertCpuProf (nodes, id2Index, id) {
+  const node = nodes[id2Index[id]]
+  const name = node.callFrame.functionName
+  const url = node.callFrame.url
+  const lineNumber = node.callFrame.lineNumber
+  const childrenLength = node.children ? node.children.length : 0
+  const converted = {
+    children: new Array(childrenLength),
+    name: `${name}${url ? ' ' + url : ''}${lineNumber > 0 ? `:${lineNumber}` : ''}`,
+    top: node.hitCount,
+    value: node.hitCount,
+    S: 0
+  }
+  if (name[0] === '(' && childrenLength === 0) {
+    // filter out (garbage collector) and (idle)
+    converted.top = 0
+    converted.value = 0
+  }
+  for (let i = 0; i < childrenLength; i++) {
+    converted.children[i] = convertCpuProf(nodes, id2Index, node.children[i])
+    converted.value += converted.children[i].value
+  }
+  return converted
+}
+
 function cpuProfileToTree (profile) {
-  const converted = convert(profile.head)
+  const converted = profile.head ?
```

Comment:
Are you sure the unique way to identify both profiles is through the `head` property?

Reply:
It's not unique ... one can also check for `profile.nodes` (the screenshot at https://github.com/davidmarkclements/0x/issues/215#issuecomment-516880388 suggests other possible keys) but I chose perhaps the simplest way by basing the check on current code.

Reply:
I'm just wondering if this is safe to use. I mean, V8 can decide to change the top structure, and it will silently fail

Reply:
Major changes to the structure would probably result in a stack trace similar to what is inside https://github.com/davidmarkclements/0x/issues/215#issue-474890398. Tbh I don't see how V8 can reasonably naturally evolve the top structure while also causing silent failure.

Bugs in 0x itself are outside the scope of this pr.